src/cavendish_particle_tracks/_set_fiducial_dialog.py

```diff
@@ -0,0 +1,536 @@
+import napari
+import numpy as np
+from qtpy.QtWidgets import (
+    QAbstractItemView,
+    QComboBox,
+    QDialog,
+    QDialogButtonBox,
+    QGridLayout,
+    QHBoxLayout,
+    QLabel,
+    QPushButton,
+    QTableWidget,
+    QTableWidgetItem,
+    QVBoxLayout,
+)
+
+from ._analysis import FIDUCIAL_BACK, FIDUCIAL_FRONT, Fiducial
+from ._calculate import (
+    corrected_length,
+    depth,
+    magnification,
+)
+
+
+class Set_Fiducial_Dialog(QDialog):
+    def __init__(self, parent):
+        # Call parent constructor
+        super().__init__(parent)
+        self.parent = parent
+
+        # ignoring ruff style here to make the array clearer
+        # fmt:off
+        self.points = [
+            [Fiducial("Front Fiducial 1 View 1"), Fiducial("Front Fiducial 1 View 2")],
+            [Fiducial("Front Fiducial 2 View 1")],
+            [Fiducial("Point View 1"), Fiducial("Point View 2")],
+            [Fiducial("Back Fiducial 1 View 1"),Fiducial("Back Fiducial 1 View 2")],
+            [Fiducial("Back Fiducial 2 View 1")]
```

Comment:
If we decide to keep the magnification separate, I guess we only need the position of two fiducial in each of the views:

```suggestion
            [Fiducial("Front Fiducial View 1"), Fiducial("Front Fiducial View 2")],
            [Fiducial("Point View 1"), Fiducial("Point View 2")],
            [Fiducial("Back Fiducial View 1"),Fiducial("Back Fiducial View 2")],
```

Reply:
This is better implemented on the stereoshift_dialog file, since they're being kept separate anyway.
I might actually just implement this here, duplicate the files and rename it to stereoshift dialog to replace the file. That way, we can keep both whilst we're still comparing the two approaches.